solr/solrj/src/java/org/apache/solr/common/SolrDocument.java

```diff
@@ -388,20 +386,47 @@ public void addChildDocuments(Collection<SolrDocument> children) {
      }
    }
 
+   @Override
+   public Map<String, Object> getChildDocumentsMap() {
+     Map<String, Object> childDocs = new HashMap<>();
+     for (Entry<String, Object> entry: _fields.entrySet()) {
+       Object value = entry.getValue();
+       if(objIsDocument(value)) {
+         childDocs.put(entry.getKey(), value);
+       }
+     }
+     return childDocs;
+   }
+
    /** Returns the list of child documents, or null if none. */
    @Override
    public List<SolrDocument> getChildDocuments() {
-     return _childDocuments;
+     List<SolrDocument> childDocs = new ArrayList<>();
+     Stream<AbstractMap.SimpleEntry<String, SolrDocument>> fields = _fields.entrySet().stream()
+         .filter(value -> value.getValue() instanceof SolrInputDocument)
+         .map(value -> new AbstractMap.SimpleEntry<>(value.getKey(), (SolrDocument) value.getValue()));
+     fields.forEach(e -> childDocs.add(e.getValue()));
+     return childDocs.size() > 0 ? childDocs: null;
    }
    
    @Override
    public boolean hasChildDocuments() {
```

Comment:
We'd probably deprecate these if we go with this overall approach.  It's too much internal cost to call this method.